application/models/team.py

```diff
@@ -0,0 +1,63 @@
+from application.errors import NotFound, NotInitialized
+from flask import current_app
+from py2neo import Graph, Node
+import uuid
+
+
+class Team:
+
+    @classmethod
+    def all(cls):
+        results = current_app.graph.run("MATCH (x:Team) RETURN x")
+        teams = []
+        for result in results:
+            team = Team(result['x']['name'])
```

Comment:
This looks weird to me... why do you use `Team` within the same class?
I mean I understand, but wouldn't it be better to separate the `Team` model and the queries?

Reply:
I'm going for an ActiveRecord like model. So you can say.: `Team.all()` and it will return all the Team Objects as an Array. Yes you could extract this to a different class but seems like upfront design to do that for just one model.
Abstracting Elasticsearch and the Graphdatabase in a useful way will be a very interesting task nontheless.